src/Compilers/Core/Portable/SourceGeneration/RunResults.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Text;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Represents the results of running a generation pass over a set of <see cref="ISourceGenerator"/>s.
+    /// </summary>
+    public readonly struct GeneratorDriverRunResult
+    {
+        internal GeneratorDriverRunResult(ImmutableArray<GeneratorRunResult> results, ImmutableArray<SyntaxTree> trees, ImmutableArray<Diagnostic> diagnostics)
+        {
+            this.Results = results;
+            this.SyntaxTrees = trees;
+            this.Diagnostics = diagnostics;
+        }
+
+        /// <summary>
+        /// The individual result of each <see cref="ISourceGenerator"/> that was run in this generator pass, one per generator.
+        /// </summary>
+        public ImmutableArray<GeneratorRunResult> Results { get; }
+
+        /// <summary>
+        /// The <see cref="Diagnostic"/>s produced by all generators run during this generation pass.
+        /// </summary>
+        /// <remarks>
+        /// This is equivalent to the union of all <see cref="GeneratorRunResult.Diagnostics"/> in <see cref="Results"/>.
+        /// </remarks>
+        public ImmutableArray<Diagnostic> Diagnostics { get; }
+
+        /// <summary>
+        /// The <see cref="SyntaxTree"/>s produced during this generation pass by parsing each <see cref="SourceText"/> added by each generator.
+        /// </summary>
+        /// <remarks>
+        /// This is equivalent to the union of all <see cref="GeneratedSourceResult.SyntaxTree"/>s in each <see cref="GeneratorRunResult.AddedSources"/> in each <see cref="Results"/>
+        /// </remarks>
+        public ImmutableArray<SyntaxTree> SyntaxTrees { get; }
+    }
+
+    /// <summary>
+    /// Represents the results of a single <see cref="ISourceGenerator"/> generation pass.
+    /// </summary>
+    public readonly struct GeneratorRunResult
+    {
+        internal GeneratorRunResult(ISourceGenerator generator, ImmutableArray<GeneratedSourceResult> addedSources, ImmutableArray<Diagnostic> diagnostics, Exception? exception)
+        {
+            this.Generator = generator;
+            this.AddedSources = addedSources;
+            this.Diagnostics = diagnostics;
+            this.Exception = exception;
+        }
+
+        /// <summary>
+        /// The <see cref="ISourceGenerator"/> that produced this result.
+        /// </summary>
+        public ISourceGenerator Generator { get; }
+
+        /// <summary>
+        /// The sources that were added by <see cref="Generator"/> during the generation pass this result represents.
+        /// </summary>
+        public ImmutableArray<GeneratedSourceResult> AddedSources { get; }
```

Comment:
I see the implementation on the generator API side of things is calling this additional sources; here it's added sources, and the type is a GeneratedSourceResult (not say an "AddedSource"). Do we want some sort of consistency here?

Reply:
Lets go with 'GeneratedSources'

---
In reply to: [465382289](https://github.com/dotnet/roslyn/pull/46296#discussion_r465382289) [](ancestors = 465382289)